src/jit/morph.cpp

```diff
@@ -7372,6 +7372,12 @@ void Compiler::fgMorphTailCall(GenTreeCall* call, void* pfnCopyArgs)
     if (call->IsVirtualStub())
     {
         GenTree* stubAddrArg = fgGetStubAddrArg(call);
+
+        // We don't need this arg to be in the normal stub register, so
+        // clear out the register assignment.
+        assert(stubAddrArg->gtRegNum == virtualStubParamInfo->GetReg());
+        stubAddrArg->ClearRegNum();
```

Comment:
Why did you do this instead of just `stubAddrArg->gtRegNum = REG_NA`?

Reply:
Doing that would leave `gtRegTag` set to `GT_REGTAG_REG`.

Reply:
I am also curious and have the same question.

Reply:
Sure. We do set `foo->gtRegNum = REG_NA` in many places. The `GT_REGTAG_NONE` thing is to indicate it's semantically unintialized, as opposed to being initialized to `REG_NA`. I would think someone might expect to read it as `REG_NA`. But if you think nobody expects to read it, then what you've done is ok.

Reply:
Hmm. Looks like `gtRegTag` is not a fully baked concept.

I'll change how I do this since it seems to diverge from what we do elsewhere.

Reply:
I'd like to see us more clearly specify the behavior of `gtRegTag`, but I believe it's been useful in the past to catch places where we used an uninitialized value here. `GT_REG_NA` is used in the register allocator to indicate both "no register was allocated here because it was optional" as well as "no register is produced by this node", and maybe that could use some clarification.

Reply:
One way in which it is "not fully baked" is that certain callers read uninitialized gtRegNum values -- that's why there are "TODO" about allowing `GT_REGTAG_NONE`. It seems the offenders are harmless, and I never tracked down and cleaned them up to not do this.